packages/flutter_tools/lib/src/base/os.dart

```diff
@@ -315,23 +323,26 @@ class _WindowsUtils extends OperatingSystemUtils {
 
   @override
   void zip(Directory data, File zipFile) {
-    final Archive archive = Archive();
-    for (final FileSystemEntity entity in data.listSync(recursive: true)) {
-      if (entity is! File) {
-        continue;
-      }
-      final File file = entity as File;
-      final String path = file.fileSystem.path.relative(file.path, from: data.path);
-      final List<int> bytes = file.readAsBytesSync();
-      archive.addFile(ArchiveFile(path, bytes.length, bytes));
-    }
-    zipFile.writeAsBytesSync(ZipEncoder().encode(archive), flush: true);
+    _processManager.runSync(<String>[
+      _activePowershell,
+      'Compress-Archive',
+      '-Path',
+      data.path,
+      '-DestinationPath',
+      zipFile.path,
+    ]);
   }
 
   @override
   void unzip(File file, Directory targetDirectory) {
-    final Archive archive = ZipDecoder().decodeBytes(file.readAsBytesSync());
-    _unpackArchive(archive, targetDirectory);
+    _processManager.runSync(<String>[
+      _activePowershell,
+      'Expand-Archive',
+      '-Path',
+      file.path,
+      '-DestinationPath',
+      targetDirectory.path,
```

Comment:
Should we pass `-Force`? I can't tell from the docs https://docs.microsoft.com/de-de/powershell/module/microsoft.powershell.archive/Expand-Archive?view=powershell-7

Reply:
From my testing locally it didn't look necessary. It is also not used in https://github.com/flutter/flutter/blob/master/bin/internal/update_dart_sdk.ps1#L81